File: main/funny-string/funny-string.cs
#define DEBUG

using System;
using System.Diagnostics.Contracts;
using System.Linq;

internal static class Solution {
    private static void Main()
    {
        foreach (var t in Enumerable.Range(0, int.Parse(Console.ReadLine()))) {
            var s = Console.ReadLine();
            var r = new string(s.Reverse().ToArray());

            Contract.Assert(s.Length > 0);
            var funny = Enumerable.Range(1, s.Length - 1)
                                  .All(j => Math.Abs(s[j] - s[j - 1])
                                         == Math.Abs(r[j] - r[j - 1]));

            Console.WriteLine(funny ? "Funny" : "Not Funny");
        }
    }
}

File: main/funny-string/funny-string.rb
gets.to_i.times do
	s = gets.chomp
	r = s.reverse

	funny = (1 .. s.size - 1).all? {|i|
		(s[i].ord - s[i - 1].ord).abs == (r[i].ord - r[i - 1].ord).abs
	}

	puts funny ? "Funny" : "Not Funny"
end
